course5/homework12/line_sorting.cc

```diff
@@ -0,0 +1,47 @@
+#include <stdexcept>
+#include <limits>
+
+#include "line_sorting.h"
+
+namespace jds {
+
+std::vector<std::string> ReadLines(std::istream & source)
+{
+	std::vector<std::string> lines;
+	std::string line;
+	while (source)
+	{
+		std::getline(source, line);
+		if (bool(source) || line.length())
+			lines.push_back(std::move(line));
+	}
+	return lines;
+}
+
+std::string FindColValue(std::string const & line, char separator, size_t column)
+{
+	size_t pos = 0;
```

Comment:
Of course. I briefly looked at `strtok()` and quickly discarded it as an option, but should have realized that if there was no tokenizer in the standard library there should have been one in Boost